mlflow/metrics/genai/model_utils.py

```diff
@@ -215,6 +218,32 @@ def _call_gateway_api(gateway_uri, payload, eval_parameters):
         )
 
 
+def _construct_payload_from_str(prompt: str, endpoint_type: str) -> Dict[str, Any]:
+    """
+    Construct the payload from the input string based on the endpoint type.
+    If the endpoint type is not specified or unsupported one, raise an exception.
+    """
+    if endpoint_type == "llm/v1/completions":
+        return {"prompt": prompt}
+    elif endpoint_type == "llm/v1/chat":
+        return {"messages": [{"role": "user", "content": prompt}]}
+    else:
+        raise MlflowException(
+            f"Unsupported endpoint type: {endpoint_type}. If string input is provided, "
+            "the endpoint type must be 'llm/v1/completions' or 'llm/v1/chat'.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
```

Comment:
We don't need to worry about `embeddings`?

Reply:
I see, `_call_deployments_api` doesn't support embeddings.

Reply:
Yup, this is dedicated for llm-as-a-judge metric now so we don't need embedding🙂